### ISSUE TITLE
how to write data cfg file?

### ISSUE THREAD

**Issue by hitbuyi** [ultralytics/yolov5#13609]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

mydata
├── dataSet
├── image
├── label
└── label_xml
where dataSet includes trainval.txt,val.txt, image is png image, label includes txt lablefiles,
how to write data cfg ymal files?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @hitbuyi, thank you for your interest in YOLOv5 🚀! Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) all the way to advanced concepts like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```
## Environments
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response. An Ultralytics engineer will review your question and assist you soon 😊

**Comment by pderrenger** [comment]
To create a YAML configuration file for your dataset structure, you'll need to specify the dataset path and split information. Here's an example based on your folder structure:

```yaml
# mydata.yaml
path: /path/to/mydata  # dataset root dir
train: dataSet/trainval.txt  # path to train images list (relative to 'path')
val: dataSet/val.txt  # path to val images list (relative to 'path')

# Classes
nc: 80  # number of classes
names: ['person', 'bicycle', 'car', ...]  # class names
```

Make sure your `trainval.txt` and `val.txt` files contain relative paths to your images in the `image` folder, and that your label files in the `label` folder follow the YOLO format with corresponding filenames to your images.

**Comment by hitbuyi** [comment]
> To create a YAML configuration file for your dataset structure, you'll need to specify the dataset path and split information. Here's an example based on your folder structure:
> 
> # mydata.yaml
> path: /path/to/mydata  # dataset root dir
> train: dataSet/trainval.txt  # path to train images list (relative to 'path')
> val: dataSet/val.txt  # path to val images list (relative to 'path')
> 
> # Classes
> nc: 80  # number of classes
> names: ['person', 'bicycle', 'car', ...]  # class names
> Make sure your `trainval.txt` and `val.txt` files contain relative paths to your images in the `image` folder, and that your label files in the `label` folder follow the YOLO format with corresponding filenames to your images.

 I think it should be `/images/ `and `/labels/`,  image files should be placed in` images `folder, and labels should be placed in` labels `folder

**Comment by pderrenger** [comment]
You're absolutely right! For YOLOv5 to work properly, your dataset structure should follow the standard convention with `images` and `labels` folders. You'll need to rename your `image` folder to `images` and your `label` folder to `labels`, then update your YAML accordingly:

```yaml
# mydata.yaml
path: /path/to/mydata  # dataset root dir
train: images/train  # or dataSet/trainval.txt if using txt files
val: images/val      # or dataSet/val.txt if using txt files

# Classes
nc: 80  # number of classes
names: ['person', 'bicycle', 'car', ...]  # class names
```

This ensures YOLOv5 can automatically find the corresponding label files in the `labels` folder for each image in the `images` folder.

**Comment by faolpmk** [comment]
Thanks a lot.

**Comment by pderrenger** [comment]
Happy to help—if anything still looks off, please share your `mydata.yaml` plus the exact `python train.py ... --data mydata.yaml` command and we can spot what YOLOv5 is resolving for `images/` and `labels/` (see the [YOLOv5 Custom Data Training Guide](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) for the expected YAML format).

**Comment by glenn-jocher** [comment]
Closing this since the dataset YAML and expected `images/` / `labels/` layout were answered above. If a current YOLOv5 `master` run still fails with a specific `data.yaml`, please open a new issue with the YAML, command, and full error output.
